framework_tvb/tvb/tests/framework/core/factory.py

```diff
@@ -240,34 +242,37 @@ def import_default_project(admin_user=None):
         return import_service.created_projects[0]
 
     @staticmethod
-    def launch_importer(importer_class, view_model, user, project_id):
-        # type: (type, ViewModel, User, int) -> None
+    def launch_importer(importer_class, view_model, user, project, sync=True):
+        # type: (type, ViewModel, User, Project, bool) -> None
         importer = ABCAdapter.build_adapter_from_class(importer_class)
-        OperationService().fire_operation(importer, user, project_id, view_model=view_model)
+        if sync:
+            TestFactory.launch_synchronously(user, project, importer, view_model)
+        else:
+            OperationService().fire_operation(importer, user, project.id, view_model=view_model)
```

Comment:
Everything looks very good to me in these changes. 
This `if sync` was puzzling, as I didn't understand it at first.

Maybe add a comment here:
- `sync = False` will do the normal flow, with Uploaders running synchronously but in a different process. This branch won't be compatible with usage in subclasses of `TransactionalTestCase` because the upload results won't be available for the unit-test running.
-  `sync = True` for usage in subclasses of `TransactionalTestCase`, as data preparation, for example. Won't test the "real" upload flow, but it is very close to that.

Reply:
ABCAdapter looks clean now after you also reviewed it in #194